src/core/core.animations.js

```diff
@@ -1,121 +1,154 @@
 'use strict';
 
-var defaults = require('./core.defaults');
-var helpers = require('../helpers/index');
+const Animator = require('./core.animator');
+const Animation = require('./core.animation');
+const helpers = require('../helpers/index');
+const defaults = require('./core.defaults');
 
 defaults._set('global', {
 	animation: {
 		duration: 1000,
 		easing: 'easeOutQuart',
+		active: {
+			duration: 400
+		},
+		resize: {
+			duration: 0
+		},
+		numbers: {
+			type: 'number',
+			properties: ['x', 'y', 'borderWidth', 'radius']
+		},
 		onProgress: helpers.noop,
 		onComplete: helpers.noop
 	}
 });
 
-module.exports = {
-	animations: [],
-	request: null,
-
-	/**
-	 * @param {Chart} chart - The chart to animate.
-	 * @param {Chart.Animation} animation - The animation that we will animate.
-	 * @param {number} duration - The animation duration in ms.
-	 * @param {boolean} lazy - if true, the chart is not marked as animating to enable more responsive interactions
-	 */
-	addAnimation: function(chart, animation, duration, lazy) {
-		var animations = this.animations;
-		var i, ilen;
-
-		animation.chart = chart;
-		animation.startTime = Date.now();
-		animation.duration = duration;
+function copyOptions(target, values) {
+	let oldOpts = target.options;
+	let newOpts = values.options;
+	if (!oldOpts || !newOpts || newOpts.$shared) {
+		return;
+	}
+	if (oldOpts.$shared) {
+		target.options = helpers.extend({}, oldOpts, newOpts, {$shared: false});
+	} else {
+		helpers.extend(oldOpts, newOpts);
+	}
+	delete values.options;
+}
+
+class Animations {
+	constructor(chart, animations) {
+		this._chart = chart;
+		this._properties = new Map();
+		this.configure(animations);
+	}
 
-		if (!lazy) {
-			chart.animating = true;
-		}
+	configure(animations) {
+		const animatedProps = this._properties;
+		const animDefaults = Object.fromEntries(Object.entries(animations).filter(({1: value}) => !helpers.isObject(value)));
 
-		for (i = 0, ilen = animations.length; i < ilen; ++i) {
-			if (animations[i].chart === chart) {
-				animations[i] = animation;
-				return;
+		for (let [key, cfg] of Object.entries(animations)) {
+			if (!helpers.isObject(cfg)) {
+				continue;
+			}
+			for (let prop of cfg.properties || [key]) {
+				// Can have only one config per animation.
+				if (!animatedProps.has(prop)) {
+					animatedProps.set(prop, helpers.extend({}, animDefaults, cfg));
+				} else if (prop === key) {
+					// Single property targetting config wins over multi-targetting.
+					animatedProps.set(prop, helpers.extend({}, animatedProps.get(prop), cfg));
+				}
 			}
 		}
+	}
 
-		animations.push(animation);
-
-		// If there are no animations queued, manually kickstart a digest, for lack of a better word
-		if (animations.length === 1) {
-			this.requestAnimationFrame();
-		}
-	},
-
-	cancelAnimation: function(chart) {
-		var index = helpers.findIndex(this.animations, function(animation) {
-			return animation.chart === chart;
-		});
+	/**
+	 * @private
```

Comment:
Will do, just commenting here also. 

If options are cacheable (= not scripted or indexed), those are shared between elements.

Those shared options are then animated separately. So having one animation object per common point option, like radius. This is mostly why the demo is so fast compared to master where every point's every property is interpolated.

This could be still improved - we could have some of the options shared and some unique to point. But lets leave that for later, if ever.

(talking about point here, because its the element that is usually numerous. applies to bar as well.)